[Bitcode] Fix constexpr expansion creating invalid PHIs

timower opened the pull request:
Fixes errors about duplicate PHI edges when the input had duplicates with constexprs in them. The constexpr translation makes new basic blocks, causing the verifier to complain about duplicate entries in PHI nodes.

nikic commented on llvm/test/Bitcode/constexpr-to-instr-dups.ll:
```diff
@@ -0,0 +1,32 @@
+; RUN: opt -expand-constant-exprs -p verify %s.bc
```
It's fine to drop the llvm-dis.

Commit:
[Bitcode] Fix constexpr expansion creating invalid PHIs

Fixes errors about duplicate PHI edges when the input had duplicates with
constexprs in them. The constexpr translation makes new basic blocks,
causing the verifier to complain about duplicate entries in PHI nodes.

nikic reviewed (approved):
LGTM
